Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -151,9 +151,10 @@ void TextureCache::Cleanup(int _frameCount)
 		{
 			iter->second->frameCount = _frameCount;
 		}
-		if (_frameCount > TEXTURE_KILL_THRESHOLD + iter->second->frameCount &&
-		    // EFB copies living on the host GPU are unrecoverable and thus shouldn't be deleted
-		    !iter->second->IsEfbCopy())
+		// Only remove EFB copies when they wouldn't be used anymore(changed hash), because EFB copies living on the
+		// host GPU are unrecoverable. Perform this check only every TEXTURE_KILL_THRESHOLD for performance reasons
+		if ((_frameCount - iter->second->frameCount) % TEXTURE_KILL_THRESHOLD == 1 && (!iter->second->IsEfbCopy() ||
```

Comment:
Also, now that I've split up the logic, wouldn't this result in non-efb copies being freed the frame after they were created? 

    ((X+1) - X) mod Y == 1